Composer/packages/client/src/components/Notifications/NotificationButton.tsx

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { css, jsx } from '@emotion/core';
+import React, { useState } from 'react';
+import { FontWeights } from '@uifabric/styling';
+import { IButtonStyles, IconButton } from 'office-ui-fabric-react/lib/Button';
+import { NeutralColors, SharedColors } from '@uifabric/fluent-theme';
+import { useRecoilValue } from 'recoil';
+import formatMessage from 'format-message';
+
+import { notificationsSelector } from '../../recoilModel/selectors/notificationsSelector';
+import { dispatcherState } from '../../recoilModel';
+
+import { NotificationPanel } from './NotificationPanel';
+
+const styles = {
+  container: css`
+    position: relative;
+  `,
+  count: (visible?: boolean) => css`
+    background-color: ${NeutralColors.white};
+    border: 2px solid ${SharedColors.cyanBlue10};
+    border-radius: 100%;
+    color: ${SharedColors.cyanBlue10};
+    font-size: 8px;
+    font-weight: ${FontWeights.bold};
+    height: 12px;
+    right: -4px;
+    position: absolute;
+    text-align: center;
+    visibility: ${visible ? 'visible' : 'hidden'};
+    width: 12px;
+  `,
+};
+
+interface NotificationButtonProps {
+  buttonStyles?: IButtonStyles;
+}
+
+const NotificationButton: React.FC<NotificationButtonProps> = ({ buttonStyles }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { deleteNotification, markNotificationAsRead } = useRecoilValue(dispatcherState);
+  const notifications = useRecoilValue(notificationsSelector);
+  const unreadNotification = notifications.filter(({ read }) => !read);
+
+  const toggleIsOpen = () => {
+    if (!isOpen) {
+      notifications.map(({ id }) => markNotificationAsRead(id));
```

Comment:
Question: as soon as one clicks the icon, notifications are set to read and hidden?

Also, given that the addNotification adds to the end of the array of notifications, users may miss unread notifications that are located below the viewport (scrolled).

Reply:
When there is a new notification, it renders as a toast. When the user opens the panel, the notifications should be cleared from the clear. Setting the notification to hidden removes it from the toaster - not the panel. A more descriptive name or a comment would be helpful here. Also, when the panel is opened, the notification is considered read.